### ISSUE TITLE
The documentation is targeting the wrong audience 

### ISSUE THREAD

**Issue by Aboghazala** [indygreg/PyOxidizer#78]
wow, i spent hours reading how great pyoxidizer is and didn't find a simple instructions how to convert foo.py to foo.exe
the documentation is targeting the wrong audience you talk "Rust" most of the time for python people,   this is a packaging tool for python, i know it is written in rust but as a user it is too much reading, 

> simple is better than complex

anyhow,  
good luck with your project, and hope to find it working, 
i am sorry if my words is dry, at the end of the day you put a lot of effort and time in this project and gave it to people for free, and my critics is solely to give you a feedback to enhance your project

**Comment by pagli17** [comment]
I totally agree with you even though you could have exposed the same concept with nicer words :)

**Comment by Aboghazala** [comment]
I found this youtube video

https://youtu.be/0_DsRQI1L8c

But I couldn’t find a way to import the modules in my app.py 
Also tried the venv option and didn’t work

**Comment by manugarri** [comment]
yeah i think this project got a lot more attention that it deserved given its development status (because its from a mozilla dev). I think a complex tool like this will take a long time to be usable.

**Comment by Luckz** [comment]
> P.S. Did you find the instruction to convert a .py to .exe? I need it as well

https://github.com/indygreg/PyOxidizer/issues/27#issuecomment-506298186

**Comment by Aboghazala** [comment]
@Luckz
For a simple app like print(“hello world”) it works, but I couldn’t make it work for a real application that required external modules to be imported

[same person comment]https://github.com/indygreg/PyOxidizer/issues/27#issuecomment-506618563

**Comment by gnprice** [comment]
> Did you find the instruction to convert a .py to .exe? I need it as well

In case it helps other folks curious how to do this: here's a small example I constructed after playing with this tool a bit today:
https://github.com/gnprice/hello-pyoxidizer/commit/66071d04435d03cdd1490c617d82c0e167d478c1

The crux of it is, in `pyoxidizer.toml`:
```
[[packaging_rule]]
type = "package-root"
path = "."
packages = ["hello"]

[[embedded_python_run]]
mode = "module"
module = "hello.hello"
```
and then make an empty `hello/__init__.py` (so `hello/` is a package directory), and stick the script at `hello/hello.py`.

---

> For a simple app like print(“hello world”) it works, but I couldn’t make it work for a real application that required external modules to be imported

I also constructed a small example that pulls in a dependency from PyPI with `pip install`:
https://github.com/gnprice/hello-pyoxidizer/commit/26b8d3fed46e3207d3899211a7299d9ab5764a00

You add
```
[[packaging_rule]]
type = "pip-install-simple"
package = "YOUR_DEPENDENCY"
```
to `pyoxidizer.toml`, and then just import the library from your own code as usual.

Note that a lot of libraries in the ecosystem don't work, though. One reason is #69 -- using `__file__`, which Python docs say is non-portable but which is pretty common.

If a particular library you want to use doesn't work, though, I think @indygreg would be glad to see details in an issue here!

---

(I agree with everyone here that it'd be helpful to have examples like these in the docs! I'm wrapping up for the evening, and may not come back to this. Please all feel free to borrow anything from my examples that's useful in sending PRs for the docs.)

**Comment by Aboghazala** [comment]
@gnprice, thanks for sharing
